### PR TITLE
Fix for onDuplicateKeyUpdate in mysql

### PR DESCRIPTION
**Pull request by rfink**
Allow onDuplicateKeyUpdate in mysql for schemas that have field names defined.

### CODE REVIEW

File: lib/dialects/mysql/query-generator.js [hunk 1]
```diff
@@ -189,7 +189,8 @@ var QueryGenerator = {
 
     if (options && options.updateOnDuplicate) {
       onDuplicateKeyUpdate += ' ON DUPLICATE KEY UPDATE ' + options.updateOnDuplicate.map(function(attr) {
-        var key = this.quoteIdentifier(attr);
+        var field = rawAttributes && rawAttributes[attr] && rawAttributes[attr].field || attr;
```
Comment (janmeier) [inline lib/dialects/mysql/query-generator.js]: `rawAttributes` should always be defined, and field will always be set, so you should be able to shorten this clause

**Comment by janmeier** [comment]
Looks good - could you add a test?

**Comment by janmeier** [comment]
ping @rfink

**Comment by rfink** [comment]
Hey guys, sorry I haven't responded.  I'll take a look tonight.  There was a reason I didn't write a test, I'll figure out why.

**Comment by rfink** [comment]
So now I remember.  The only specific unit test for updateOnDuplicate works without models (it's only on the query generator side) - test/integration/dialects/mysql/query-generator.test.js.  Any recommendations on how I would go about adding one?

**Comment by janmeier** [comment]
@rfink Some sql unit tests in test/unit/sql should be fine

**Comment by rfink** [comment]
Okay, I'll work on something.  Thanks.

**Comment by rfink** [comment]
I've added a bulkCreate test to the insert test/unit/sql/insert test, I had to skip mssql because it breaks on matching up field aliases to their hash key (specifically, when trying to access the autoIncrement property).  It seems to work as expected for MySQL, and the onDuplicateKeyUpdate gets ignored elsewhere (as it should).  Let me know if there's anything else I can do.

**Comment by rfink** [comment]
Eek, small typo in postgres.  Fixing and testing now...

**Comment by rfink** [comment]
The failing test was a timeout on an unrelated test.  Can you re-run?

**Comment by janmeier** [comment]
Looks good @rfink - If you address my previous comment and squash + rebase, this should be good to merge

**Comment by rfink** [comment]
Ah, hadn't saw that.  I will work on getting this done.  Thanks.

**Comment by rfink** [comment]
I admit I'm haven't done a ton of rebasing, but let me know if this is what you are looking for.  Thanks in advance.

**Comment by janmeier** [comment]
@rfink Somehow you managed to get the messages for all four commits into the squashed one :D 

No worries, I rebased and pushed in https://github.com/sequelize/sequelize/commit/da57c5b5878d079724bd81e1a6ab89d23efdc6ff

**Comment by janmeier** [comment]
Thanks! :)

**Pull request closed without merging**